src/Core/LoginConfig/ConfigBuilder/TemplateData/TemplateData.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\LoginConfig\ConfigBuilder\TemplateData;
+
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class TemplateData
+{
+    public function __construct(
+        public readonly string $key,
+        public readonly string $snippetKey,
+        public readonly string $icon,
+        public readonly string $class,
+        public readonly string $url,
+        public readonly string $random,
+    ) {
+    }
+
+    /**
+     * @return array{key: string, snippet_key: string, icon: string, class: string, url: string}
+     */
+    public function toArray(): array
```

Comment:
Maybe it helps to:
```php
class TemplateData implements \JsonSerializable
```
so it should be possible to pass that object directly to `JsonResponse` 🤔 

There is also the `AssignArrayTrait`, just take a look at `\Shopware\Core\Framework\Struct\Struct`

*Might need adjustments on different DTO classes you have here, just added the comment here because it looks provides a way to convert into an array